packages/app/src/cli/services/context.ts

```diff
@@ -161,31 +173,44 @@ async function ensureIncludeConfigOnDeploy({
     includeConfigOnDeploy: previousIncludeConfigOnDeploy,
   })
 
-  if (force || previousIncludeConfigOnDeploy !== undefined) return
+  if (force || previousIncludeConfigOnDeploy === true) return
   await promptIncludeConfigOnDeploy({
     appDirectory: app.directory,
     localApp: app,
   })
 }
 
+async function removeIncludeConfigOnDeployField(localApp: AppInterface) {
+  const patch = {build: {include_config_on_deploy: undefined}}
+  await patchAppConfigurationFile({path: localApp.configuration.path, patch, schema: localApp.configSchema})
+  const message = {
+    headline: `Configuration is now included on deploy`,
+    body: [
+      `The \`include_config_on_deploy\` field has been removed from your configuration file and is now enabled by default.`,
+    ],
+    link: {
+      label: 'See Shopify CLI documentation.',
+      url: 'https://shopify.dev/docs/apps/build/cli-for-apps/app-configuration#build',
+    },
+  }
+  return renderInfo(message)
+}
+
 async function promptIncludeConfigOnDeploy(options: ShouldOrPromptIncludeConfigDeployOptions) {
   const shouldIncludeConfigDeploy = await includeConfigOnDeployPrompt(options.localApp.configuration.path)
   const localConfiguration = options.localApp.configuration as CurrentAppConfiguration
-  localConfiguration.build = {
-    ...localConfiguration.build,
-    include_config_on_deploy: shouldIncludeConfigDeploy,
-  }
-
   const patch = {build: {include_config_on_deploy: shouldIncludeConfigDeploy}}
   await patchAppConfigurationFile({path: localConfiguration.path, patch, schema: options.localApp.configSchema})
   await metadata.addPublicMetadata(() => ({cmd_deploy_confirm_include_config_used: shouldIncludeConfigDeploy}))
 }
 
 function includeConfigOnDeployPrompt(configPath: string): Promise<boolean> {
   return renderConfirmationPrompt({
-    message: `Include \`${basename(configPath)}\` configuration on \`deploy\`?`,
-    confirmationMessage: 'Yes, always (Recommended)',
-    cancellationMessage: 'No, never',
+    message: `Include \`${basename(
+      configPath,
+    )}\` configuration on \`deploy\`? The \`include_config_on_deploy\` field will be deprecated soon and enabled by default, so we recommend to do it now.`,
```

Comment:
```suggestion
    )}\` configuration on \`deploy\`? Soon, this will no longer be an option, and configuration will need to be included on deploy every time.`,
```